$MirOS: ports/devel/glib1/patches/patch-configure_in,v 1.4 2005/07/19 16:45:51 bsiegert Exp $
--- configure.in.orig	2001-03-15 16:37:37.000000000 +0100
+++ configure.in	2009-07-07 20:08:41.000000000 +0200
@@ -48,10 +48,10 @@ AC_SUBST(GLIB_INTERFACE_AGE)
 AC_SUBST(GLIB_BINARY_AGE)
 
 # libtool versioning
-LT_RELEASE=$GLIB_MAJOR_VERSION.$GLIB_MINOR_VERSION
-LT_CURRENT=`expr $GLIB_MICRO_VERSION - $GLIB_INTERFACE_AGE`
+LT_RELEASE=
+LT_CURRENT=`expr $GLIB_MAJOR_VERSION + $GLIB_MINOR_VERSION`
 LT_REVISION=$GLIB_INTERFACE_AGE
-LT_AGE=`expr $GLIB_BINARY_AGE - $GLIB_INTERFACE_AGE`
+LT_AGE=$GLIB_MINOR_VERSION
 AC_SUBST(LT_RELEASE)
 AC_SUBST(LT_CURRENT)
 AC_SUBST(LT_REVISION)
@@ -544,6 +544,13 @@ if test -z "$G_MODULE_IMPL"; then
 		G_MODULE_IMPL=G_MODULE_IMPL_DLD
 	,)
 fi
+
+if test "x`uname -s`" = "xOpenBSD" -a "x$enable_shared" = "xyes"; then
+  if test -z "`echo __ELF__ | $CC -E - | grep __ELF__`" || test "$host_os-$host_cpu" = "openbsd2.8-powerpc"; then
+    G_MODULE_LDFLAGS="-Wl,-E"
+  fi
+fi
+
 dnl *** additional checks for G_MODULE_IMPL_DL
 if test "$G_MODULE_IMPL" = "G_MODULE_IMPL_DL"; then
 	case "$host_os" in
@@ -800,6 +807,12 @@ G_THREAD_CFLAGS="$G_THREAD_CFLAGS -D_REE
 		# FreeBSD 2.2.x shiped with gcc 2.7.2.x, which doesn't support
 		# -mthreads flag.
 		;;
+	*-mirbsd* | *-openbsd*)
+		G_THREAD_CFLAGS="$G_THREAD_CFLAGS -D_THREAD_SAFE"
+		  if test "x$G_THREAD_LIBS" = "x-lc_r"; then
+		    G_THREAD_LIBS="-pthread"
+		  fi
+		;;
     esac
 
     # if we are not finding the ctime_r function, then we probably are
@@ -915,8 +928,6 @@ AC_DEFINE_UNQUOTED(G_THREAD_SOURCE,"gthr
 AC_SUBST(G_THREAD_CFLAGS)
 AC_SUBST(G_THREAD_LIBS)
 
-CFLAGS="$CFLAGS $G_THREAD_CFLAGS"
-
 dnl **********************************************
 dnl *** GDefaultMutex setup and initialization ***
 dnl **********************************************
